lighthouse-core/audits/non-composited-animations.js

```diff
@@ -0,0 +1,138 @@
+  /** [ICU Syntax] Label identifying the number of animations that are not composited. */
+  displayValue: `{itemCount, plural,
+  =1 {# animated element found}
+  other {# animated elements found}
+  }`,
+  /** Name of a compositor failure reason where the CSS property being animated is not supported on the compositor. */
+  unsupportedCSSProperty: 'Unsupported CSS Property',
+};
+
+const str_ = i18n.createMessageInstanceIdFn(__filename, UIStrings);
+
+/**
+ * Each failure reason is represented by a bit flag. The bit shift operator '<<' is used to define which bit corresponds to each failure reason.
+ * https://source.chromium.org/search?q=f:compositor_animations.h%20%22enum%20FailureReason%22
+ * @type {{flag: number, text: string}[]}
+ */
+const ACTIONABLE_FAILURE_REASONS = [
+  {
+    flag: 1 << 13,
+    text: str_(UIStrings.unsupportedCSSProperty),
+  },
+];
+
+/**
+ * Return list of actionable failure reasons and a boolean if some reasons are not actionable.
+ * Each flag is a number with a single bit set to 1 in the position corresponding to a failure reason.
+ * We can check if a specific bit is true in the failure coding using bitwise and '&' with the flag.
+ * @param {number} failureCode
+ * @return {string[]}
+ */
+function getActionableFailureReasons(failureCode) {
+  return ACTIONABLE_FAILURE_REASONS
+    .filter(reason => failureCode & reason.flag)
+    .map(reason => reason.text);
+}
+
+class NonCompositedAnimations extends Audit {
+  /**
+   * @return {LH.Audit.Meta}
+   */
+  static get meta() {
+    return {
+      id: 'non-composited-animations',
+      scoreDisplayMode: Audit.SCORING_MODES.INFORMATIVE,
+      title: str_(UIStrings.title),
+      description: str_(UIStrings.description),
+      requiredArtifacts: ['TraceElements', 'HostUserAgent'],
+    };
+  }
+
+  /**
+   * @param {LH.Artifacts} artifacts
+   * @return {Promise<LH.Audit.Product>}
+   */
+  static async audit(artifacts) {
+    // COMPAT: This audit requires m86
+    const match = artifacts.HostUserAgent.match(/Chrome\/(\d+)/);
+    if (!match || Number(match[1]) < 86) {
+      return {
+        score: 1,
+        notApplicable: true,
+      };
+    }
+
+    /** @type LH.Audit.Details.TableItem[] */
+    const results = [];
+    artifacts.TraceElements.forEach(element => {
+      if (element.traceEventType !== 'animation') return;
+      /** @type LH.Audit.Details.NodeValue */
+      const node = {
+        type: 'node',
+        path: element.devtoolsNodePath,
+        selector: element.selector,
+        nodeLabel: element.nodeLabel,
+        snippet: element.snippet,
+      };
+
+      const animations = element.animations || [];
+      const failureReasons = new Set();
+      for (const {name, failureReasonsMask} of animations) {
+        if (!failureReasonsMask) continue;
+        for (const failureReason of getActionableFailureReasons(failureReasonsMask)) {
+          failureReasons.add({failureReason, animation: name});
```

Comment:
ah, drat hang on @adamraine this set isn't doing it's job anymore with the object-form

also brings up a problem when there are no animations with an identifiable name, maybe we use `Unknown` and use a string as a key here

Reply:
\~Technically\~ an animation could have `Unknown` as it's display name.  I think `undefined` can be used as a key for a map so I will use that.